### DISCUSSION TITLE
ESP8266 support

### DISCUSSION THREAD

**Discussion by techniciansoft** [tobozo/WiFiManagerTz#4] [Q&A]
Hi!
Thank you for your library!
Your library is perfect for my project, just what I need. The only problem - I can't compile example for ESP8266 Board.
Could you help me to adopt this library for ESP8266, please?

**Comment by tobozo** [comment]
hi, thanks for your feedback :+1: 

WiFiManagerTz takes advantage of the snmp layer and posix timezones native support from esp-idf and is limited to esp32, and adapting to esp8266 will require more than just replacing the snmp_* function calls. 

Problems to solve:
1) the function `handleRoute()` builds the HTML page without checking for available heap, esp8266 will be OOM 
2) esp8266 sdk doesn't have builtin support for `sntp_set_time_sync_notification_cb()` and `sntp_set_sync_interval()`

Possible solutions:
1) => use buffered response in `handleRoute()`, or send as chunks
2) => use an external NTP library

Ideally `handleRoute()` should drop `getTemplate()` calls and let you maintain your own layout elements in PROGMEM or SPIFFS, so that you can stream the content of the `<select>` menus without even using the heap.

**Reply by techniciansoft** [reply]
Thank you very much for the clarification!
Аctually have to solve the first problem.
The second problem in my projects is already solved as you said, with the help of external libraries 
https://forum.arduino.cc/t/esp8266-ntp-tz-dst-example-simplified/632223

line`static esp8266::polledTimeout::periodicMs showTimeNow(20000);` set the polling interval and `settimeofday_cb(time_is_set_scheduled);` for call back function...

**Reply by tobozo** [reply]
interesting, for some reasom I though `<PolledTimeout.h>` was an external library.

there's a third problem: persistence, currently it's using Preferences.h with esp32, and should use EEPROM.h with esp8266, this means more changes in prefs.hpp/cpp
